sdks/v4-sdk/src/entities/pool.ts

```diff
@@ -316,8 +317,9 @@ export class Pool {
     }
   }
 
-  private nonImpactfulHook(): boolean {
-    // TODO: reference chain specific hook addresses or patterns that do not impact swaps
-    return this.hooks === ADDRESS_ZERO
+  private hookHasSwapPermissions(): boolean {
+    return (
+      Hook.hasPermission(this.hooks, HookOptions.BeforeSwap) || Hook.hasPermission(this.hooks, HookOptions.AfterSwap)
```

Comment:
do we need to check `BeforeSwapReturnsDelta` and `AfterSwapReturnsDelta`?

Reply:
No, because these options only get hit if either beforeSwap or AfterSwap are true. (before/afterSwapReturnsDelta cannot be true if beforeSwap or afterSwap are false)

https://github.com/Uniswap/v4-core/blob/main/src/libraries/Hooks.sol#L265

Reply:
I can leave a comment